src/instantiator.js

```diff
@@ -65,7 +83,9 @@ function instantiate(schema) {
       // Visit each property.
       for (var property in obj.properties) {
         if (obj.properties.hasOwnProperty(property)) {
-          visit(obj.properties[property], property, data[name]);
+          if ((!options.requiredPropertiesOnly) || (options.requiredPropertiesOnly && isPropertyRequired(property, obj.required))) {
```

Comment:
This line is too long, I think moving it to a function `shouldVisit(property, obj, options)` would be best, since we'll probably want to add conditions and options in the future.